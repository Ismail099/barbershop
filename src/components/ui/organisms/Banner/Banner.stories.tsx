import { Story, Meta } from '@storybook/react';

import { Banner, BannerProps } from './Banner';

export default {
  title: 'organisms/Banner',
  component: Banner,
  args: {
    image: '/assets/img/image-banner-1.jpg'
  },
  parameters: {
    backgrounds: {
      default: 'dark',
      values: [{ name: 'dark', value: '#121212' }]
    },
    layout: 'fullscreen'
  }
} as Meta;

export const Basic: Story<BannerProps> = (args) => <Banner {...args} />;
