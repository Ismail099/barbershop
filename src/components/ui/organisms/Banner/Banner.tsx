import * as S from './Banner.styles';

import Fade from 'react-reveal/Fade';

import { Title } from '~atoms/Title/Title';
import { Subtitle } from '~atoms/Subtitle/Subtitle';

export type BannerProps = {
  image: string;
};

export const Banner = ({ image }: BannerProps) => {
  return (
    <S.Banner src={image}>
      <Fade delay={300} bottom>
        <Title size="large">Seja o Primeiro a Saber</Title>
      </Fade>
      <Fade delay={400} bottom>
        <Subtitle size="large">Em breve sua Barbearia favorita terÃ¡ uma nova plataforma inaugurada!</Subtitle>
      </Fade>
    </S.Banner>
  );
};
