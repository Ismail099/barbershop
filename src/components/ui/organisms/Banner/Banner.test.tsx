import { renderWithTheme } from '~src/utils/tests/helpers';

import { Banner } from './Banner';

describe('<Banner />', () => {
  it('Deve renderizar o componente <Banner />', () => {
    //Arrange
    const { container } = renderWithTheme(<Banner image="/assets/img/image-banner-1.jpg" />);
    //Assert
    expect(container.firstChild).toMatchSnapshot();
  });
});
