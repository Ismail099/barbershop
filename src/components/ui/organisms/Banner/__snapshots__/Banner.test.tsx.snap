// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> Deve renderizar o componente <Banner /> 1`] = `
.c0 {
  width: 100%;
  height: 100vh;
  background-image: url(/assets/img/image-banner-1.jpg);
  background-position: center;
  background-size: cover;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  color: #F71812;
  font-family: Dancing Script;
  font-weight: 500;
  font-size: 3.5rem;
}

.c2 {
  max-width: 105rem;
  color: #FFFFFF;
  font-family: Abril Fatface;
  font-weight: 400;
  font-size: 3.0rem;
}

@media (min-width:768px) {
  .c1 {
    font-size: 5.0rem;
  }
}

@media (min-width:768px) {
  .c2 {
    font-size: 6.0rem;
  }
}

<div
  class="c0"
  src="/assets/img/image-banner-1.jpg"
>
  <div
    class="react-reveal"
  >
    <h2
      class="c1"
    >
      Seja o Primeiro a Saber
    </h2>
  </div>
  <div
    class="react-reveal"
  >
    <h2
      class="c2"
    >
      Em breve sua Barbearia favorita ter√° uma nova plataforma inaugurada!
    </h2>
  </div>
</div>
`;
